x/upgrade/internal/keeper/keeper.go

```diff
@@ -130,3 +140,66 @@ func (k Keeper) ApplyUpgrade(ctx sdk.Context, plan types.Plan) {
 func (k Keeper) IsSkipHeight(height int64) bool {
 	return k.skipUpgradeHeights[height]
 }
+
+// DumpUpgradeInfoToDisk adds plan height to upgrade-info.json
+func (k Keeper) DumpUpgradeInfoToDisk(height int64, name string) error {
+	upgradeInfoFilePath, err := k.GetUpgradeInfoPath()
+	if err != nil {
+		return err
+	}
+
+	upgradeInfo := store.UpgradeInfo{
+		Name:   name,
+		Height: height,
+	}
+	info, err := json.Marshal(upgradeInfo)
+
+	if err != nil {
+		return err
+	}
+
+	return ioutil.WriteFile(upgradeInfoFilePath, info, 0644)
+}
+
+// GetUpgradeInfoPath returns the upgrade info file path
+func (k Keeper) GetUpgradeInfoPath() (string, error) {
+	upgradeInfoFileDir := path.Join(k.GetHomeDir(), "data")
+
+	if _, err := os.Stat(upgradeInfoFileDir); os.IsNotExist(err) {
+		err = os.Mkdir(upgradeInfoFileDir, os.ModePerm)
+		if err != nil {
+			return "", err
+		}
+		return filepath.Join(upgradeInfoFileDir, UpgradeInfoFileName), nil
+	}
+
+	return filepath.Join(upgradeInfoFileDir, UpgradeInfoFileName), nil
+}
+
+// GetHomeDir returns the height at which the given upgrade was executed
+func (k Keeper) GetHomeDir() string {
```

Comment:
Does this need to be exposed/used anywhere outside the keeper?

Reply:
Not really. I will change that